packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart

```diff
@@ -14,9 +16,174 @@ part 'sk_payment_queue_wrapper.g.dart';
 /// The payment queue contains payment related operations. It communicates with App Store and presents
 /// a user interface for the user to process and authorize the payment.
 class SKPaymentQueueWrapper {
+  SKTransactionObserverWrapper _observer;
+
+  /// Returns the default payment queue.
+  ///
+  /// We do not support instantiating a custom payment queue, hence the singleton. However, you can
+  /// override the observer
+  factory SKPaymentQueueWrapper() {
+    return _singleton;
+  }
+
+  static final SKPaymentQueueWrapper _singleton = new SKPaymentQueueWrapper._();
+
+  SKPaymentQueueWrapper._() {
+    callbackChannel.setMethodCallHandler(_handleObserverCallbacks);
+  }
+
   /// Calls [`-[SKPaymentQueue canMakePayments:]`](https://developer.apple.com/documentation/storekit/skpaymentqueue/1506139-canmakepayments?language=objc).
   static Future<bool> canMakePayments() async =>
       await channel.invokeMethod('-[SKPaymentQueue canMakePayments:]');
+
+  /// Sets a transaction observer to listen to all the transaction events of the payment queue.
+  ///
+  /// You have to have a transaction observer to be added to the payment queue to handle the
+  /// payment follow.
+  /// You must set the observer right when App launches to avoid missing callback when your user
+  /// started a purchase flow from the App Store.
+  void setTransactionObserver(SKTransactionObserverWrapper observer) {
+    _observer = observer;
+  }
+
+  /// Adds a payment object to the payment queue.
+  ///
+  /// Prior to this call, you have to make sure At least one [SKTransactionObserverWrapper] should be added to the payment queue
+  /// using [addTransactionObserver]. You also have to make sure the [SKProductWrapper] of the payment has been fetched using [SKRequestMaker.startProductRequest].
+  /// Each payment will generate a [SKPaymentTransactionWrapper]. After a payment is being added to the payment queue, the [SKTransactionObserverWrapper] is responsible to handle
+  /// the transaction that is generated by the payment.
+  /// The [productIdentifier] must match one of the product that is returned in [SKRequestMaker.startProductRequest].
+  /// to test the payment in the [sandbox](https://developer.apple.com/apple-pay/sandbox-testing/).
+  Future<void> addPayment(SKPaymentWrapper payment) async {
```

Comment:
We should link to the official dev docs on the related methods for all of these too.